test/hpke_test.c

```diff
@@ -0,0 +1,1432 @@
+                                                   senderpublen, privp,
+                                                   infop, infolen)))
+                        overallresult = 0;
+                    /* throw in a call with a too-short clearlen */
+                    clearlen = 15;
+                    if (!TEST_false(OSSL_HPKE_open(rctx, clear, &clearlen,
+                                                   aadp, aadlen, cipher,
+                                                   cipherlen)))
+                        overallresult = 0;
+                    /* fix up real clearlen again */
+                    clearlen = OSSL_HPKE_TSTSIZE;
+                    if (!TEST_true(OSSL_HPKE_open(rctx, clear, &clearlen,
+                                                  aadp, aadlen, cipher,
+                                                  cipherlen)))
+                        overallresult = 0;
+                    OSSL_HPKE_CTX_free(rctx);
+                    EVP_PKEY_free(privp);
+                    privp = NULL;
+                    /* check output */
+                    if (!TEST_mem_eq(clear, clearlen, plain, plainlen)) {
+                        overallresult = 0;
+                    }
+                    if (privp) {
+                        EVP_PKEY_free(privp);
+                        privp = NULL;
+                    }
+                }
+            }
+            EVP_PKEY_free(authpriv);
+        }
+    }
+    return overallresult;
+}
+
+/**
+ * @brief check roundtrip for export
+ * @return 1 for success, other otherwise
+ */
+static int test_hpke_export(void)
+{
+    EVP_PKEY *privp = NULL;
+    unsigned char pub[OSSL_HPKE_TSTSIZE];
+    size_t publen = sizeof(pub);
+    int hpke_mode = OSSL_HPKE_MODE_BASE;
+    OSSL_HPKE_SUITE hpke_suite = OSSL_HPKE_SUITE_DEFAULT;
+    OSSL_HPKE_CTX *ctx = NULL;
+    OSSL_HPKE_CTX *rctx = NULL;
+    unsigned char exp[32];
+    unsigned char exp2[32];
+    unsigned char rexp[32];
+    unsigned char rexp2[32];
+    unsigned char plain[] = "quick brown fox";
+    size_t plainlen = sizeof(plain);
+    unsigned char enc[OSSL_HPKE_TSTSIZE];
+    size_t enclen = sizeof(enc);
+    unsigned char cipher[OSSL_HPKE_TSTSIZE];
+    size_t cipherlen = sizeof(cipher);
+    unsigned char clear[OSSL_HPKE_TSTSIZE];
+    size_t clearlen = sizeof(clear);
+    char * estr = "foo";
+
+    if (!TEST_true(OSSL_HPKE_keygen(hpke_suite, pub, &publen, &privp,
+                                    NULL, 0, testctx, NULL)))
+        goto end;
+    if (!TEST_ptr(ctx = OSSL_HPKE_CTX_new(hpke_mode, hpke_suite,
+                                          testctx, NULL)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_encap(ctx, enc, &enclen, pub, publen, NULL, 0)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_seal(ctx, cipher, &cipherlen, NULL, 0,
+                                  plain, plainlen)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_export(ctx, exp, 32,
```

Comment:
`sizeof(exp)` instead of 32. Similar comment everywhere that 32 appears in this test below here.

Reply:
fair 'nuff, done.